grakn-core/src/main/java/ai/grakn/graql/GraqlConverter.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.graql;
+
+import ai.grakn.concept.Concept;
+import ai.grakn.graql.admin.Answer;
+
+import javax.annotation.CheckReturnValue;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * An interface for converting objects in Graql responses into other formats.
+ *
+ * @param <Builder> An intermediate builder type that can be changed into a {@link T}
+ * @param <T> The type to convert into
+ *
+ * @author Felix Chapman
+ */
+public interface GraqlConverter<Builder, T> {
```

Comment:
The distinction between `GraqlConverter` and `Printer` is kind of vague. Maybe even just changing the to be more explicit to something like:

```
An interface for converting any objects (e.g. Answers, Concepts, etc . . . ) in Graql responses into specific Types `T`.
```